### PR TITLE
Add ProcessInfoByPids API

### PR DESCRIPTION
**Pull request by marcov**
ProcessInfoByPids API is used to to retrieve process information from a list of user specified processes. 

The sample application has been updated to use it as an option.

Implements #32

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ This library aims to make things a bit more comfortable, especially for containe
  - `psgo.ProcessInfo(descriptors []string) ([][]string, error)`
    - ProcessInfo returns the process information of all processes in the current mount namespace. The input descriptors must be a slice of supported AIX format descriptors in the normal form or in the code form, if supported.  If the input descriptor slice is empty, the `psgo.DefaultDescriptors` are used. The return value contains the string slice of process data, one per process.
 
+ - `psgo.ProcessInfoByPids(pids []string, descriptors []string) ([][]string, error)`
+   - ProcessInfoByPids returns the process information of a set of processes in the current mount namespace. The input pids must be a slice of PIDs. The input descriptors must be a slice of supported AIX format descriptors in the normal form or in the code form, if supported.  If the input descriptor slice is empty, the `psgo.DefaultDescriptors` are used. The return value contains the string slice of process data, one per process.
```
Comment (vrothberg) [inline README.md]: I'd prefer to say that it's similar to `psgo.ProcessInfo` but will only return the data of the specified pids. It's less to read and easier to figure the differences among the functions.

File: README.md [hunk 2]
```diff
@@ -28,6 +31,17 @@ root         4       2       0.000    6h3m27.678701852s    ?        0s
 root         6       2       0.000    6h3m27.678999508s    ?        0s          [mm_percpu_wq]
 ```
 
+### Listing processes
+You can use the `--pids` flag to restrict `psgo` output to a subset of processes. This options accepts a list of comma separate process IDs and will return exactly the same kind of information per process as the default output.
```
Comment (vrothberg) [inline README.md]: small typo: s/options/option/

File: sample/sample.go [hunk 3]
```diff
@@ -14,11 +14,13 @@ import (
 func main() {
 	var (
 		descriptors []string
+		pidsList    []string
 		data        [][]string
 		err         error
 	)
 
 	pid := flag.String("pid", "", "join mount namespace of the process ID")
+	pids := flag.String("pids", "", "Comma separated list of process IDs to retrieve")
```
Comment (vrothberg) [inline sample/sample.go]: s/Comma/comma/

File: sample/sample.go [hunk 4]
```diff
@@ -33,11 +35,24 @@ func main() {
 		descriptors = strings.Split(*format, ",")
 	}
 
+	if *pid != "" && *pids != "" {
+		fmt.Printf("You can't pass both --pid and --pids options")
```
Comment (vrothberg) [inline sample/sample.go]: Can we `os.Exit()` here?

File: README.md (hunk 1 above)
Comment (vrothberg) [reply README.md]: We also need to mention what happens when the slice is empty.

Comment (marcov) [reply README.md]: When the slice is empty  a slice of size [1][n]  is returned containing the n format descriptor headers and no process information. That's ok or would you rather return nothing at all?

Comment (vrothberg) [reply README.md]: Actually, there is another case when no matching pids could be found: "When the specified slice of pids is empty or no matching processes could be found, only the header without further data is returned."

**Comment by vrothberg** [comment]
@marcov Looks good to me. Can you squash the commits before we merge the PR?

**Comment by marcov** [comment]
Squashed!

**Comment by vrothberg** [comment]
> Squashed!

Great, thanks! Once the tests are green again, I'll merge.

@rhatdan, that's similar to what we need for `podman pod top` but for the same namespace. I'll start working on the API for `pod top` now.

**Pull request merged**